1. Print first 10 natural numbers using while loop

In [ ]:
i=1
while i>0 and i <11:
  print(i,"\n")
  i+=1

1 

2 

3 

4 

5 

6 

7 

8 

9 

10 



2. Display numbers from -10 to -1 using for loop

In [ ]:
for i in range(-10,0):
  print(i)

-10
-9
-8
-7
-6
-5
-4
-3
-2
-1


3.Display a message “Done” after successful execution of for loop

In [ ]:
for i in range(5):
  print(i)
print("Done!")

0
1
2
3
4
Done!


In [ ]:
4. Calculate the sum of all numbers from 1 to N

In [ ]:
n=int(input())
res=0
for i in range(1,n+1):
  res+=i
print(res)

10
55


## **5. Print multiplication table of a given number**

In [ ]:
n=2
for i in range(1,11):
 print(n*i)

2
4
6
8
10
12
14
16
18
20


# **6. Calculate the cube of all numbers from 1 to a given number**

In [ ]:
n=10
for i in range(1,n+1):
  print(f"The cube of {i} is {i**3}")

The cube of 1 is 1
The cube of 2 is 8
The cube of 3 is 27
The cube of 4 is 64
The cube of 5 is 125
The cube of 6 is 216
The cube of 7 is 343
The cube of 8 is 512
The cube of 9 is 729
The cube of 10 is 1000


# **7. Display numbers from a list using a loop**

Given a list of numbers, iterate through it and print numbers that satisfy these conditions:


The number must be divisible by five.
If the number is greater than 150, skip it and move to the next.
If the number is greater than 500, stop the loop entirely.

In [ ]:
n= [12, 75, 150, 180, 145, 525, 50]
for i in n:
   if i>500:
    break

   if i>150:
    continue


   if i%5==0:
    print(i)


75
150
145


# **8. Count occurrences of a specific element in a list**

Given a list of numbers, use a loop to count how many times a specific number (e.g., 10) appears.

In [ ]:
a=[10, 20, 10, 30, 10, 40, 50]
n=10
c=0
for i in a:
  if n==i:
    c+=1
print(c)


3


# **9. Print elements from a list present at odd index positions**

Given a Python list, use a loop to print only the elements that are located at odd index positions (index 1, 3, 5, etc.).

In [ ]:
a=[10,20,30,40,50,6,7]
for i in range(1,len(a),2):
      print(a[i])

20
40
6


# **10.Print list in reverse order using a loop**


In [ ]:
Given a list, iterate it in reverse order and print each element.

In [ ]:
a=[7,6,5,4,3,2,1]
for i in range(len(a)-1,-1,-1):
  print(a[i])

1
2
3
4
5
6
7


# **11. Reverse a string using a for loop (no slicing)**

Write a program that takes a string and reverses it using a for loop. While Python’s [::-1] shortcut is famous, reversing a string manually is a classic way to understand how sequences are constructed.

In [ ]:
a="sister"
print(a[::-1])

retsis


In [ ]:
a="welcome"
for i in range(len(a)-1,-1,-1):
 print(a[i],end="")

emoclew

In [ ]:
original_str = "Python"
reversed_str = ""

for char in original_str:
    # Prepending: Put the new character before the existing ones
    reversed_str = char + reversed_str

print(f"Original: {original_str}")
print(f"Reversed: {reversed_str}")

Original: Python
Reversed: nohtyP


# **12. Count vowels and consonants in a sentence**

Write a program that counts the total number of vowels and consonants in a given sentence, ignoring spaces and special characters.

In [ ]:
a="welcome!"
v=0
c=0
for i in a:
  if i.isalpha():
   if i in ["a","e","i","o","u"]:
    v+=1
   else:
    c+=1
print(f"The total number of vowels is {v} and consonants is {c}")

The total number of vowels is 3 and consonants is 4


# **13. Count total number of digits in a number**

Write a program to count the total number of digits in a given integer using a while loop.

In [ ]:
num=987675
c=0
while num!=0:
  num=num//10
  c+=1
print(c)


6


# **14. Reverse an integer number**

Write a program to reverse a given integer number (e.g., 76542 should become 24567).

In [ ]:
a=67752
rev=0
while a>0:
 d=a%10
 rev=(rev*10)+d
 a=a//10
print(rev)

25776


# **15. Find largest and smallest digit in a number**

Write a program to find the largest and smallest digit within a given integer (e.g., in 75869, the largest is 9 and the smallest is 5).

In [ ]:
a=int(input())
largest=0
smallest=9
while a>0:
 num=a%10
 if num>largest:
  largest=num

 if num<smallest:
  smallest=num

 a=a//10

print(f"In {a}, the largest is {largest} and the smallest is {smallest}")

47851432
In 0, the largest is 8 and the smallest is 1


# **16. Check if a number is a palindrome**

Write a program to check if a given number is a palindrome. A palindrome number is a number that remains the same when its digits are reversed (e.g., 121, 343).

In [ ]:
a=int(input())
d=a
c=0
while a>0:
  b=a%10
  c=c*10+b
  a=a//10
if c==d:
    print("True")
else:
    print("Not a palindrome")


50505
True


# **17. Find factorial of a number**

Write a program to use a loop to find the factorial of a given number (e.g., 5!). The factorial of N is the product of all integers from 1 to N.

In [ ]:
a=int(input())
result=1
if a==0:
  print("Factorial of 0 is 1")
elif a<0:
  print("Factorial does not exist for negative numbers")
else:
   for i in range(1,a+1):
    res=i*result
    result=res
   print(f"The factorial of {a} is {result}")

-1
Factorial does not exist for negative numbers


# **18. Collatz Conjecture: Generate a sequence until it reaches 1**

The Collatz conjecture states that if you start with any positive integer n, and if n is even, divide it by 2; if n is odd, multiply it by 3 and add 1. Repeat the process. The sequence will always eventually reach 1. Write a program to print this sequence for a given number.

In [ ]:
num=int(input())
print(num,end="")

while num!=1:
  if num%2==0:
     num=num//2
  else :
    num=num*3+1
  print(f",{num}",end="")


5
5,16,8,4,2,1

# **19. Armstrong Number Check**

 Write a program to check if a number is an Armstrong number. An Armstrong number (for a 3-digit number) is an integer such that the sum of the cubes of its digits is equal to the number itself (e.g., 153 = 1^3 + 5^3 + 3^3).

In [ ]:
a=int(input())
b=str(a)
power=len(b)
res=0

for i in b:
  res+=int(i)**power

if res==a:
  print(f"{a} is an Armstrong number")

else:
  print(f"{a} is not an Armstrong Number")



1634
1634 is an Armstrong number


# **20. Print right-angled triangle Number Pattern using a Loop**

 Write a program to print a right-angled triangle pattern where each row contains increasing numbers up to the row number

In [ ]:
for i in range(1,6):
  for j in range(1,i+1):
    print(j,end="")
  print("")

1
12
123
1234
12345


In [ ]:
for i in range(1,6):
  for j in range(1,i+1):
    print(j,end=" ")
  print(" ")


1  
1 2  
1 2 3  
1 2 3 4  
1 2 3 4 5  


# **21. Print the decreasing pattern**

Write a program to use for loop to print the following reverse number pattern:

In [ ]:
for i in range(5,0,-1):
  for j in range(i,0,-1):
    print(j,end=" ")
  print("")

5 4 3 2 1 
4 3 2 1 
3 2 1 
2 1 
1 


In [ ]:
chr(97)

TypeError: 'int' object is not callable

# **22. Print the alternate numbers pattern**

Write a program to print a pattern of alternate numbers from 1 to 20 (e.g., 1, 3, 5…).

# **23. Print Alphabet pyramid (A, BB, CCC) pattern**

Write a program to print a triangle pattern where each row consists of the same letter,
and the letter changes (increments) with each new row.

In [ ]:
chr(65)

'A'

In [ ]:
a=65
n=int(input())
for i in range(0,n):
  for j in range(0,i+1):
    print(chr(a+i),end=" ")
  print("")

5
A 
B B 
C C C 
D D D D 
E E E E E 


# **24. Hollow square pattern**

Print a 5*5 square of stars where the middle is empty, leaving only the border.

In [ ]:
n=int(input())
for i in range(0,n):
  for j in range(0,n):
    if i==0 or i==n-1 or j==0 or j==n-1:
      print("*",end=" ")
    else:
      print(" ",end=" ")
  print()



7
* * * * * * * 
*           * 
*           * 
*           * 
*           * 
*           * 
* * * * * * * 


# **25. Print pyramid pattern of stars**

 Write a program to print the following pattern using nested loops:


In [ ]:
n=int(input())
for i in range(0,n):
  for j in range(0,i+1):
    print("*",end=" ")
  print()
for m in range(n-1,0,-1):
  for o in range(0,m):
    print("*",end=" ")
  print()

5
* 
* * 
* * * 
* * * * 
* * * * * 
* * * * 
* * * 
* * 
* 


# **26. Print full multiplication table (1 to 10)**

In [ ]:
Write a program to print the full multiplication table from 1 to 10 in a grid format.

In [ ]:
n=int(input())
for i in range(1,n+1):
  for j in range(1,n+1):
    print(i*j,end=" ")
  print("")

10
1 2 3 4 5 6 7 8 9 10 
2 4 6 8 10 12 14 16 18 20 
3 6 9 12 15 18 21 24 27 30 
4 8 12 16 20 24 28 32 36 40 
5 10 15 20 25 30 35 40 45 50 
6 12 18 24 30 36 42 48 54 60 
7 14 21 28 35 42 49 56 63 70 
8 16 24 32 40 48 56 64 72 80 
9 18 27 36 45 54 63 72 81 90 
10 20 30 40 50 60 70 80 90 100 


# **27. List Cumulative Sum: Each element is the sum of all previous**

Given a list of numbers, create a new list where each element is the sum of all elements from the original list up to that position.

In [ ]:
a=[10,20,30,40]
b=[]
tot=0
for i in a:
  tot+=i
  b.append(tot)
print(b)

[10, 30, 60, 100]


# **28. Dictionary Filter: Extract pairs where value exceeds a threshold.**

Given a dictionary of student scores, create a new dictionary that only includes students who scored above a certain threshold (e.g., 75).

In [ ]:
dic={"Phy":77,
     "Math":72,
     "Che":34,
     "Bio":56}
new_dic={}
for i in dic.values():
  if i>75:
    new_dic["top_score"]=i
print(new_dic)


{'top_score': 77}


# **29. Find common elements (Intersection) using loop**

 Given two lists, find the elements that appear in both. Do not use Python’s built-in set().intersection() method.

In [ ]:
l1=[1,2,3,4]
l2=[2,3,9,7]
res=[]
for i in l1:
  if i in l2:
    res.append(i)
print(f"Common elements : {res}")

Common elements : [2, 3]


# **30. Remove duplicates without set**

Write a program to remove all duplicate values from a list using a loop, maintaining the original order of elements.

In [ ]:
l3=[1,2,3,4,2,1,5]
l4=[]
for i in l3:
  if i not in l4:
      l4.append(i)
  else:
    continue
l4


[1, 2, 3, 4, 5]

# **31. Even/Odd Segregation: Move evens to front, odds to back**

Given a list of integers, move all even numbers to the beginning of the list and all odd numbers to the end.

In [ ]:
a=[1,2,3,4,5,6,7,8,9]
b=[]
c=[]
for i in a:
  if i%2!=0:
    b.append(i)
  else:
    c.append(i)
c.extend(b)
print(c)

[2, 4, 6, 8, [1, 3, 5, 7, 9]]


# **32. List Rotation: Rotate elements left by k positions**

Given a list and an integer k, rotate the list to the left by k positions. For example, if k=2, the first two elements move to the end of the list.

In [48]:
nums=[1,2,3,4,5]
k=2
a=nums[k:]
b=nums[:k]
result=a+b
result

[3, 4, 5, 1, 2]

In [47]:
nums=[1,2,3,4,5]
k=2

for _ in range(k):
  result=nums.pop(0)
  nums.append(result)
print(nums)

[3, 4, 5, 1, 2]


# **33. Word frequency counter**

Write a program to count the frequency of each word in a given string.

In [75]:
text="I am a girl I have a girl friend"
words=text.split()

freq={}

for i in words:
  if i in freq:
    freq[i]+=1

  else:
    freq[i]=1

print(freq)

{'I': 2, 'am': 1, 'a': 2, 'girl': 2, 'have': 1, 'friend': 1}


# **34. Display fibonacci series up to 10 terms**

Write a program to display the Fibonacci sequence up to 10 terms. The sequence starts with 0 and 1, and each subsequent number is the sum of the two preceding ones.

In [137]:
n1=0
n2=1
res=0
for i in range(0,10):
  print(n1,end=" ")
  res=n1+n2
  n1=n2
  n2=res

0 1 1 2 3 5 8 13 21 34 

# **35. Perfect number check**

Write a program to check if a number is a “Perfect Number.” A perfect number is a positive integer that is equal to the sum of its proper divisors (excluding the number itself). For example, 6 is perfect because 1 + 2 + 3 = 6.

In [144]:
num=15
divisor_Sum=0

for i in range(1,num//2+1):
  if num%i==0:
    divisor_Sum+=i


if divisor_Sum==num:
  print(f"The number{num} is a perfect number")

else:
  print(f"The number {num} is not a perfect number")


The number 15 is not a perfect number


# **36. Binary to decimal conversion using loop**

Manually convert a binary string (e.g., "1101") into its decimal integer equivalent using a loop. Do not use int(binary, 2).

In [164]:
a=str(1101)
pow=0
res=0
for i in a[::-1]:
  res+=(int(i)*(2**pow))
  pow+=1
print(f"The decimal integer of binary {a} is {res}")

The decimal integer of binary 1101 is 13


# **37. Display all prime numbers within a range**

Write a program to display all prime numbers within a range (e.g., 25 to 50). A prime number is a natural number greater than 1 that is not a product of two smaller natural numbers.

In [174]:
st=25
end=50

for i in range(st,end+1):
  if i>1:
      for j in range(2,i):
        if (i%j)==0:
          break

      else:
        print(i)


29
31
37
41
43
47


# **38. Find the sum of the series up to n terms**

Write a program to calculate the sum of the series 2 + 22 + 222 + 2222 + …. up to N terms. For example, if n=5, the series is 2 + 22 + 222 + 2222 + 22222.

In [197]:
a=5
num=2
num_sum=0
for i in range(0,a):
  num_sum+=num
  num=num*10+2
print(num_sum)


24690


# ** 39. flatten a nested list using loops**

Given a nested list (a list containing other lists), write a program to “flatten” it into a single list containing all the individual elements.

In [207]:
a=[[10, 20], [30, 40], [50, 60]]
b=[]

for i in a:
  for j in i:
    b.append(j)
print(b)

[10, 20, 30, 40, 50, 60]


# **40. Nested list search (2D matrix coordinates)**

 Given a 2D list (matrix), find the row and column index of a target value.

In [225]:
matrix = [[10, 20], [30, 40], [50, 60]]
target = 30

for row_id,row in enumerate(matrix):
  for col_id,col in enumerate(row):
    if target==col:
      print(row_id,col_id)

1 0
